server/gameconnection.py

```diff
@@ -362,7 +363,14 @@ async def handle_operation_complete(
                 self.game.leaderboard_saved = True
 
     async def handle_json_stats(self, stats: str):
-        self.game.report_army_stats(stats)
+        try:
+            self.game.report_army_stats(stats)
+        except json.JSONDecodeError:
+            self._logger.warning(
+                "Malformed game stats reported by %s: '...%s'",
+                self._player.login,
+                stats[-20:]
```

Comment:
I'm obviously not familiar with this codebase, but `stats[-20:]` seem quite ambiguous. What is special about the last 20 elements of this object? Maybe some comments would be useful here.

Reply:
Just that when these errors happens, it seems like it's always due to truncated data. Having the last few characters just lets you confirm that by seeing if the json structure is terminated or not.